docs/policy/memtier.md

```diff
@@ -1,70 +1,314 @@
-    on some special-purpose computing systems.
+Opting in to exclusive allocation happens by opting out from shared allocation,
+and opting out from exclusive allocation happens by opting in to shared
+allocation.
 
-In order to configure a container to use a certain memory type, use the
-`memory-type.cri-resource-manager.intel.com` effective annotation in the pod
-spec. For example, to make `container1` request both `PMEM` and `DRAM` memory
-types, you could use pod metadata such as this:
+A container can opt in to or opt out from isolated exclusive CPU core
+allocation using the following annotation.
 
-```
+```yaml
 metadata:
   annotations:
-    memory-type.cri-resource-manager.intel.com/container.container1: dram,pmem
+    # opt in container $C1 to isolted exclusive CPU core allocation
+    prefer-isolated-cpus.cri-resource-manager.intel.com/container.$C1: "true"
+    # opt in the whole pod to isolated exclusive CPU core allocation
+    prefer-isolated-cpus.cri-resource-manager.intel.com/pod: "true"
+    # selectively opt out container $C2 from isolated exclusive CPU core allocation
+    prefer-isolated-cpus.cri-resource-manager.intel.com/container.$C2: "false"
 ```
 
-Alternatively, you can use the following deprecated syntax to achieve the same,
-but support for this syntax is subject to be dropped in a future release:
+These annotations have no effect on containers which are not eligible for
+exclusive allocation.
 
-```
+### Implicit Hardware Topology Hints
+
+`CRI Resource Manager` automatically generates HW `Topology Hints` for devices
+assigned to a container, prior to handing the container off to the active policy
+for resource allocation. The `memtier` policy is hint-aware and normally takes
+topology hints into account when picking the best a pool to allocate resources.
+Hints indicate optimal `HW locality` for device access and they can alter
+significantly which pool gets picked for a container.
+
+Since device topology hints are implicitly generated, there are cases where one
+would like the policy to disregard them altogether. For instance, when a local
+volume is used by a container but not in any performance critical manner.
+
+Containers can be annotated to opt out from and selectively opt in to hint-aware
+pool selection using the following annotations.
+
+```yaml
 metadata:
   annotations:
-    cri-resource-manager.intel.com/memory-type: |
-      container1: dram,pmem
+    # only disregard hints for container $C1
+    topologyhints.cri-resource-manager.intel.com/container.$C1: "false"
+    # disregard hints for all containers by default
+    topologyhints.cri-resource-manager.intel.com/pod: "false"
+    # but take hints into account for container $C2
+    topologyhints.cri-resource-manager.intel.com/container.$C2: "true"
 ```
 
-The `memtier` policy will then aim to allocate resources from a topology
-node which can satisfy the memory requirements.
+While this might change in the future, currently implicit topology hint
```

Comment:
"While this might change in the future, currently implicit topology hint..."
->
"Topology hint..."

(I'd prefer documenting only the current behavior and avoid speculating on what could change... unless we already know for sure something is going to change soon.)

Reply:
Removed.